packages/backend/src/managers/GPUManager.ts

```diff
@@ -60,4 +79,100 @@ export class GPUManager extends Publisher<IGPUInfo[]> implements Disposable {
+  }
+
+  protected parseNvidiaCDI(stdout: string): ContainerDeviceInterface {
+    const containerDeviceInterface: unknown = load(stdout);
+    if (!containerDeviceInterface || typeof containerDeviceInterface !== 'object')
+      throw new Error('malformed output nvidia CDI output');
+    if (!('cdiVersion' in containerDeviceInterface)) throw new Error('missing cdiVersion in nvidia CDI');
+    if (containerDeviceInterface.cdiVersion !== '0.3.0')
+      throw new Error('invalid cdiVersion: expected 0.3.0 received containerDeviceInterface.cdiVersion');
+
+    if (!('kind' in containerDeviceInterface)) throw new Error('missing kind in nvidia CDI');
+    if (typeof containerDeviceInterface.kind !== 'string') throw new Error('malformed kind in nvidia CDI');
+
+    if (!('devices' in containerDeviceInterface)) throw new Error('missing devices in nvidia CDI');
+    if (!Array.isArray(containerDeviceInterface.devices)) throw new Error('devices is malformed in nvidia CDI');
+
+    return {
+      cdiVersion: containerDeviceInterface.cdiVersion,
+      kind: containerDeviceInterface.kind,
+      devices: containerDeviceInterface.devices,
+    };
+  }
+
+  /**
+   * This method will parse the `/etc/cdi/nvidia.yaml` in the available ContainerProviderConnection
+   * @protected
+   */
+  protected async getNvidiaCDI(connection: ContainerProviderConnection): Promise<ContainerDeviceInterface> {
+    if (connection.vmType) {
+      // if vmType is defined we are working with virtual machine so we need to SSH in it
+      const { stdout, stderr } = await this.podman.executeSSH(connection, ['cat', '/etc/cdi/nvidia.yaml']);
+      if (stderr.length > 0) throw new Error(stderr);
+      return this.parseNvidiaCDI(stdout);
+    }
+
+    if (!env.isLinux) {
+      throw new Error('cannot determine the environment to read nvidia CDI file');
+    }
+
+    // if vmType is undefined on linux system we are working with podman native
+    const info = await stat('/etc/cdi/nvidia.yaml');
+    if (!info.isFile()) throw new Error('invalid /etc/cdi/nvidia.yaml file');
+
+    const content = await readFile('/etc/cdi/nvidia.yaml', { encoding: 'utf8' });
+    return this.parseNvidiaCDI(content);
+  }
+
+  /**
+   * see https://github.com/cncf-tags/container-device-interface
+   * @param connection
+   */
+  public async getGPUContainerDeviceInterface(
+    connection: ContainerProviderConnection,
+  ): Promise<ContainerDeviceInterface> {
+    const gpus = this.getAll();
+    // ensure at least one GPU is available
+    if (gpus.length === 0) {
+      throw new Error('no gpu has been detected');
+    }
+
+    // ensure all GPU(s) are NVIDIA vendor
+    if (gpus.some(gpu => gpu.vendor !== GPUVendor.NVIDIA)) {
+      throw new Error('cannot get container device interface for non-nvidia GPU(s)');
+    }
+
+    // check nvidia-ctk version
+    const { version } = await this.getNvidiaContainerToolKitVersion(connection);
+    console.log('nvidia-ctk version', version);
```

Comment:
Should we keep it ?

Reply:
Yeah, in a follow-up PR I want to use telemetry, but I want to be a separate issue as discussion will probably happen :)